File: backend/src/adapters/rest/app/models/usuario.rb
class Usuario < ApplicationRecord
    # Solo tiene un cliente asociado
    has_one :cliente, dependent: :destroy
    # Solo tiene un arquitecto asociado
    has_one :arquitecto, dependent: :destroy
    # Validaciones
    validates :nombre, :apellido, :email, :estado, :password, :rol, :fecha_registro, :foto_perfil, presence: true
    # El email debe ser único
    validates :email, uniqueness: true
end

File: backend/websocket/src/domain/entities/Usuario.ts
export class Usuario {
  constructor(
    public readonly id_usuario: number,
    public nombre: string,
    public apellido: string,
    public email: string,
    public estado_cuenta: string,
    public rol: string,
    public fecha_registro: Date,
    public foto_perfil: string | null
  ) {}
}
